src/libs/sys-util-rs/src/cpu.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2019-2021 Alibaba Cloud
+// Copyright (c) 2019-2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// Test whether two CPU sets are equal.
+pub fn is_cpu_list_equal(left: &[u32], right: &[u32]) -> bool {
+    if left.len() != right.len() {
+        return false;
+    }
+
+    let mut l = left.to_vec();
+    let mut r = right.to_vec();
+    l.sort_unstable();
+    r.sort_unstable();
+    l == r
+}
+
+/// Convert CPU quota and period into milli-CPUs
+///
+/// If quota is -1, it means the CPU resource request is unconstrained.
+pub fn calculate_milli_cpus(quota: i64, period: u64) -> (u32, bool) {
+    if quota >= 0 && period != 0 {
+        ((quota as u64 * 1000 / period) as u32, false)
+    } else {
+        (0, quota == -1)
```

Comment:
It might be nicer to create an `enum` with branches for the "constrained" and "unconstrained" versions, rather than using a boolean tuple.

Reply:
refactored for better abstraction. Now implementation is in kata-types/src/cpu.rs.